clang/test/SemaSYCL/lambda_implicit_capture_this.cpp

```diff
@@ -0,0 +1,28 @@
+// RUN: %clang_cc1 -I %S/Inputs -fsycl -fsycl-is-device -verify %s
+
+#include <sycl.hpp>
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel(Func kernelFunc) {
+  kernelFunc();
+}
+
+class Class {
+public:
+  Class() : member(1) {}
+  void function();
+  int member;
+};
+
+void Class::function() {
+  kernel<class kernel_wrapper>(
+      [=]() {
+        int acc[1] = {5};
+        acc[0] *= member; // expected-error{{implicit capture of 'this' with a capture default of '=' is not allowed for kernel functions in SYCL 1.2.1}}
```

Comment:
I spent some time to actually understand how `this` was actually captured. Can we also add a clarifying note?

Reply:
Hi @Fznamznon , Thanks for the comment. Can you please let me know what we can mention in the clarifying note? Thanks.

Reply:
I wanted some note like
`'this' is captured through usage of `member` field inside the lambda`
Although, I don't see a good way to implement it now, so, you can skip this comment.